src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/RandomNumberGeneratorImplementation.cs

```diff
@@ -2,57 +2,82 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Diagnostics;
+
 namespace System.Security.Cryptography
 {
     internal sealed partial class RandomNumberGeneratorImplementation : RandomNumberGenerator
     {
-        public override void GetNonZeroBytes(byte[] data)
+        public override void GetBytes(byte[] data)
         {
-            if (data == null)
-                throw new ArgumentNullException(nameof(data));
+            if (data == null) throw new ArgumentNullException(nameof(data));
+
+            GetBytesInternal(data, 0, data.Length);
+        }
 
-            GetBytes(data);
+        public override void GetBytes(byte[] data, int offset, int count)
+        {
+            if (data == null) throw new ArgumentNullException(nameof(data));
+            if (offset < 0) throw new ArgumentOutOfRangeException(nameof(offset), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (count < 0) throw new ArgumentOutOfRangeException(nameof(count), SR.ArgumentOutOfRange_NeedNonNegNum);
+            if (offset + count > data.Length) throw new ArgumentException(SR.Argument_InvalidOffLen);
```

Comment:
Rewrite to `if (count > data.Length - offset)` to avoid integer overflow.  (`offset` has already been validated to be non-negative, so the subtraction is safe)